### PR TITLE
Updated `ProgressIndicator` to M3

### LINKED ISSUE #111449: Update `ProgressIndicator` to support Material 3
As part of the [Material 3 specification](https://m3.material.io/components/progress-indicators/overview) there are updates to progress indicator's  visuals:

unnamed

We should update the `ProgressIndicator` widgets to support this Material 3 look and tokens.

### PR DESCRIPTION
**Pull request by QuncCccccc**
Part of: #91605

Updated the `CircularProgressIndicator` widget and `LinearProgressIndicator` widget with support for [Material Design 3](https://m3.material.io/components/progress-indicators/overview).

Screen Shot 2022-09-21 at 7 01 55 PM
Screen Shot 2022-09-21 at 7 02 32 PM

In order to use the `ProgressIndicator` with the new Material 3 defaults, turn on the `useMaterial3` flag in the `ThemeData`:

```dart
  return MaterialApp(
    theme: ThemeData(useMaterial3: true),
    // ...
  );
```

Fixes #111449

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 1]
```diff
@@ -580,29 +585,44 @@ class _CircularProgressIndicatorState extends State<CircularProgressIndicator> w
...
-            valueColor: widget._getValueColor(context),
-            value: widget.value, // may be null
-            headValue: headValue, // remaining arguments are ignored if widget.value is not null
-            tailValue: tailValue,
-            offsetValue: offsetValue,
-            rotationValue: rotationValue,
-            strokeWidth: widget.strokeWidth,
-          ),
+    Widget progressIndicator = Container(
+      constraints: const BoxConstraints(
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/progress_indicator.dart:594]: Perhaps you could specify `alignment: Alignment.center` here and make constraints a property of the theme. Then the M3 value for the constraints would be tight and the M2 constraints would be as specified here.

**Review by HansMuller — APPROVED** [review]
LGTM

**Review by guidezpl — APPROVED** [review]

File: packages/flutter/lib/src/material/progress_indicator.dart (hunk 1 above)
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/progress_indicator.dart:594]: Thanks for the suggestion! The M3 size is used to make sure that the ProgressIndicator is 48x48, the circle itself should still be 36x36:)

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 2]
```diff
@@ -580,29 +585,44 @@ class _CircularProgressIndicatorState extends State<CircularProgressIndicator> w
...
+          offsetValue: offsetValue,
+          rotationValue: rotationValue,
+          strokeWidth: widget.strokeWidth,
         ),
       ),
     );
+
+    if (Theme.of(context).useMaterial3) {
+      progressIndicator = SizedBox(
+        height: _CircularProgressIndicatorDefaultsM3.circularProgressIndicatorSize,
```
Comment (akshdeep-singh) [inline packages/flutter/lib/src/material/progress_indicator.dart:614]: This makes it impossible to make the circular indicator larger than this size!

**Comment by akshdeep-singh** [comment]
@QuncCccccc I am not able to make `CircularProgressIndicator` larger than its default size.

**Pull request merged**